hashintel/hash
# Handle unexpected property type suggester errors in the frontend

## 🌟 What is the purpose of this PR?




This PR adds basic error handling to the HASH FE for the property type suggester on the create entity type page. If any error is thrown, a generic error message is thrown with the ability to retry generating the property types.

## 🔗 Related links





- [Add error handling to property type suggester](https://app.asana.com/0/1204260760205594/1204461641771566/f) _(internal)_
- [Add "New" and "Existing" labels to property type suggester](https://app.asana.com/0/1204260760205594/1204461641771570/f) _(internal)_

## 🔍 What does this change?




- modifies the GQL resolver for calling AI agents to throw errors as GQL errors.

## 🚀 Has this modified a publishable library?




This PR:


## 📜 Does this require a change to the docs?



Not that I'm aware of.

## ⚠️ Known issues



- need to fix python linting error which wasn't caught previously

## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



1. Checkout the branch
2. Add an `Exception` to `/apps/hash-agents/app/agents/generate-entity-type-property-types/__main__.py`, for example by adding `raise Exception("Test Error")` to line 21
3. Use the create entity type page to try and generate property types
4. Observe the error state, and click retry. The generation should be retried, and will fail again due to the previous exception

## 📹 Demo



(You will want to skip through the middle section where I'm restarting the docker environment so that the python changes take effect)

https://user-images.githubusercontent.com/42802102/234626107-c05e09d7-1443-4b6d-856c-dae1b377407b.mp4

At the request of @nonparibus, I've also added "Existing" and "New" tags to the property types to indicate whether they already existed or not:
<img width="603" alt="image" src="https://user-images.githubusercontent.com/42802102/234633488-ca416374-4b98-4be5-8a86-5bbe8d0389bf.png">
